mmv1/third_party/terraform/services/storagetransfer/resource_storage_transfer_job.go.tmpl

```diff
@@ -120,14 +143,44 @@ func ResourceStorageTransferJob() *schema.Resource {
 					},
 				},
 			},
+			"replication_spec": {
+				Type:          schema.TypeList,
+				MaxItems:      1,
+				Optional:      true,
+				ConflictsWith: []string{"transfer_spec", "schedule"},
+				ExactlyOneOf: []string{"transfer_spec", "replication_spec"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"object_conditions": objectConditionsSchema(replicationSpecObjectConditionsKeys),
+						"transfer_options":  transferOptionsSchema(replicationSpecTransferOptionsKeys),
+						"gcs_data_sink": {
+							Type:         schema.TypeList,
+							Optional:     true,
+							MaxItems:     1,
+							Elem:         gcsDataSchema(),
+							ExactlyOneOf: replicationSpecDataSinkKeys,
+							Description:  `A Google Cloud Storage data sink.`,
+						},
+						"gcs_data_source": {
+							Type:         schema.TypeList,
+							Optional:     true,
+							MaxItems:     1,
+							Elem:         gcsDataSchema(),
+							ExactlyOneOf: replicationSpecDataSourceKeys,
+							Description:  `A Google Cloud Storage data source.`,
+						},
+					},
+				},
+				Description: `Replication specification.`,
+			},
 			"transfer_spec": {
 				Type:     schema.TypeList,
-				Required: true,
+				Optional: true,
 				MaxItems: 1,
```

Comment:
I think it will be good to add `ConflictsWith` and `ExactlyOneOf` conditions here as well to make it explicit for each fields.

Reply:
Acknowledge